packages/server/graphql/public/types/AddedNotification.ts

```diff
@@ -0,0 +1,18 @@
+import {getUserId} from '../../../utils/authorization'
+import {AddedNotificationResolvers} from '../resolverTypes'
+
+export type AddedNotificationSource = {addedNotificationId: string}
+
+const AddedNotification: AddedNotificationResolvers = {
+  addedNotification: async ({addedNotificationId}, _args: unknown, {dataLoader, authToken}) => {
+    const viewerId = getUserId(authToken)
+    const notification = (await dataLoader.get('notifications').load(addedNotificationId)) as any
+    if (notification.userId === viewerId) {
+      return notification
+    } else {
+      return null
```

Comment:
-1 this auth check shouldn't be necessary because the `publish` call is only sending 1 message to 1 user. if we do hit the null condition, that tells us we're sending an unnecessary message to the pubsub, and we should try to avoid that! no sense in having the server do all that work just to send `null` to the client.

if we make that change, we can make the return type non-null